test/integration/datasets/DatasetsRepository.test.ts

```diff
@@ -81,20 +83,61 @@ describe('DatasetsRepository', () => {
       const actualDatasetCitation = await sut.getDatasetCitation(
         TestConstants.TEST_CREATED_DATASET_ID,
         latestVersionId,
+        false,
       );
       expect(typeof actualDatasetCitation).toBe('string');
     });
 
     test('should return error when dataset does not exist', async () => {
       let error: ReadError = undefined;
-
-      await sut.getDatasetCitation(nonExistentTestDatasetId, latestVersionId).catch((e) => (error = e));
+      await sut.getDatasetCitation(nonExistentTestDatasetId, latestVersionId,false).catch((e) => (error = e));
 
       assert.match(
         error.message,
         `There was an error when reading the resource. Reason was: [404] Dataset with ID ${nonExistentTestDatasetId} not found.`,
       );
     });
+    test('should return citation when dataset is deaccessioned', async () => {
+      let createdDatasetId = undefined;
+      await createDatasetViaApi()
+          .then((response) => (createdDatasetId = response.data.data.id))
+          .catch(() => {
+            assert.fail('Error while creating test Dataset');
+          });
+      // We publish the new test dataset we can deaccession it
+      await publishDatasetViaApi(createdDatasetId)
+          .then()
+          .catch(() => {
+            assert.fail('Error while publishing test Dataset');
+          });
+      
+      await waitForNoLocks(createdDatasetId, 10)
+      let locks = await sut.getDatasetLocks(createdDatasetId);
+      const maxTries = 10;
+      let tries = 0;
+      while (locks.length > 0 && tries < maxTries) {
+        await new Promise(resolve => setTimeout(resolve, 1000));
+        locks = await sut.getDatasetLocks(createdDatasetId);
+        tries++
+      }
+      if (tries >= maxTries && locks.length > 0) {
+        assert.fail('Error while waiting for locks to be released');
+      }
```

Comment:
If the waitForNoLocks helper function was implemented shouldn't we remove this code?
```suggestion
      await waitForNoLocks(createdDatasetId, 10)
```